jws.go

```diff
@@ -361,35 +361,44 @@ func (parsed *rawJSONWebSignature) sanitized(signatureAlgorithms []SignatureAlgo
 	return obj, nil
 }
 
+const tokenDelim = "."
+
 // parseSignedCompact parses a message in compact format.
 func parseSignedCompact(
 	input string,
 	payload []byte,
 	signatureAlgorithms []SignatureAlgorithm,
 ) (*JSONWebSignature, error) {
-	// Three parts is two separators
-	if strings.Count(input, ".") != 2 {
+	protected, s, ok := strings.Cut(input, tokenDelim)
+	if !ok { // no period found
+		return nil, fmt.Errorf("go-jose/go-jose: compact JWS format must have three parts")
+	}
+	claims, s, ok := strings.Cut(s, tokenDelim)
+	if !ok { // only one period found
+		return nil, fmt.Errorf("go-jose/go-jose: compact JWS format must have three parts")
+	}
+	sig, _, ok := strings.Cut(s, tokenDelim)
+	if ok { // three periods found
```

Comment:
The third `Cut` here seems unnecessary, right?

If a period was found, we disregard the results and return an error.

If no period was found, `sig` is equal to `s`.

I think we can skip the third `Cut` and just search for a third dot and return error.

```suggestion
	claims, sig, ok := strings.Cut(s, tokenDelim)
	if !ok { // only one period found
		return nil, fmt.Errorf("go-jose/go-jose: compact JWS format must have three parts")
	}
	if strings.ContainsRune(sig, '.') { // too many periods found
```

Reply:
Done, thank you. Maybe we should do something similar also in `ParseEncryptedCompact`